validator/docs/schemas.md

```diff
@@ -0,0 +1,193 @@
+```
+
+To further customize the deployment of your application, you can add the following sections to the configuration file.
+
+## Table of Contents
+1. [Generic Configuration](#generic-configuration)
+2. [Kubernetes Configuration](#kubernetes-configuration)
+
+
+## Generic Configuration
+
+### Processes
+
+A project can contain multiple processes such as workers, web processes and more.
+The structure to define a process is the following one:
+
+```yaml
+# Defining a simple process
+processes:
+  - name: process-name        # Process name - required - maxLength: 58
+    is_cronjob: false         # Defining a simple process
+    start_command: npm start  # Command executed when the pod is started
+# Defining a scheduled process
+  - name: cronjob-name
+    is_cronjob: true
+    start_command: npm run worker
+```
+
+### Crons
+Crons are tasks that run at a specific time or interval.
+You can use them to automate tasks, commonly as: reporting, backups, maintenance, emails and more.
+
+You can define the crons configuration with the following structure.
+
+```yaml
+crons:
+  my-cronjob: # Cronjob to define 
+              # it must be the same name as defined in processes
+    concurrency_policy: 'Forbid' # Allow, Forbid or Replace concurrency of cronjobs, STRING - Required. 
+    schedule: '* * * * *' # STRING - Required
+    suspend: false        # BOOLEAN
+```
+
+**_Note_**: You can read more about how to write a cron, the different concurrency policies and more features in the [official documentation](https://kubernetes.io/docs/tasks/job/automated-tasks-with-cron-jobs/#writing-a-cron-job-spec)
+
+### Dependencies
+
+This section can be used when your application depends or makes use of other applications. In this section you can specify the list dependencies used by your project. 
+
+```yaml
+dependencies:
+  - api-v1
+  - voice-service
+  - payment-service
+  # .. and more
+  # if your application does not have dependencies, you can omit this section
+```
+
+## Kubernetes Configuration 
+
+### Scales
+When you deploy an application in a Kubernetes environment, you define how many replicas you would like to run. 
+Each replica represents a Kubernetes Pod that encapsulates your application container. 
+You can also increase, or reduce the number of replicas in the given limit.
+
+```yaml
+scales:
+  my-process:         # Process to scale
+    min-replicas: 1   # Minimal number of pods, INTEGER - Required
+    max-replicas: 10  # Minimal number of pods, INTEGER - Required
+    targetCPUUtilizationPercentage: 80 # Percentage of CPU which will trigger a horizontal scaling when it is exceeded, INTEGER, 0 <= x <= 100 - Required
+  another-process: 
+    min-replicas: 1
+    max-replicas: 5
+    targetCPUUtilizationPercentage: 95
+```
+
+### Resources
+This structure defines the resources to be allocated for your application.
+
+```yaml
+resources:
+  my-resource:  # Name of the resource to define
+    limit:
+      cpu:      # STRING / NUMBER
```

Comment:
Why would it be a string for CPU? 🤔

Reply:
`'50'`, 50 both are valid according to the official doc 😮